File: backend/src/types/Usuario.ts
export interface Usuario {
  id: string;
  nome: string;
  email: string;
  senha: string;
  role: 'admin' | 'user';
}

export interface UsuarioLogin {
  email: string;
  senha: string;
}

export interface UsuarioResponse {
  id: string;
  nome: string;
  email: string;
  role: 'admin' | 'user';
} 

File: backend_dotnet/src/ViberLounge.Domain/Entitites/Usuario.cs
using ViberLounge.Entities;
using System.ComponentModel.DataAnnotations;

namespace ViberLounge.Domain.Entities
{
    public class Usuario : BaseEntity
    {
        [Required]
        public string? Nome { get; set; }
        [Required]
        public string? Email { get; set; }
        [Required]
        public string? Senha { get; set; }
        [Required]
        public string? Role { get; set; } = "USER";
        public virtual ICollection<Venda> Vendas { get; set; } = new HashSet<Venda>();
        public virtual ICollection<VendaCancelada> VendasCanceladas { get; set; } = new HashSet<VendaCancelada>();
    }

}